cdap-ui/app/services/constants.js

```diff
@@ -71,7 +71,7 @@ angular.module(PKG.name + '.services')
             'MISSING-REQUIRED-FIELDS': 'is missing required fields',
             'NO-SOURCE-FOUND': 'Please add a source to your pipeline',
             'MISSING-NAME': 'Pipeline name is missing.',
-            'INVALID-NAME': 'Pipeline names can only contain alphanumeric (\'a-z A-Z 0-9\') and underscore ( \'_\') characters. Please remove any other characters.',
+            'INVALID-NAME': 'Pipeline names can only contain alphanumeric (\'a-z A-Z 0-9\'), underscore (\'_\'), and hyphen (\'-\') characters. Please remove any other characters.',
```

Comment:
Yes I believe so. @elfenheart just created a pipeline with names just having underscores and hyphens. :)